CMakeLists.txt

```diff
@@ -239,6 +239,8 @@ foreach (File ${ExtraFiles})
 	install(FILES ${File} DESTINATION ${Dir})
 endforeach()
 
+install(DIRECTORY ${LocalConf} DESTINATION ${LocalConf} FILES_MATCHING PATTERN "*.ini")
+
```

Comment:
Right now, the Linux build script does not install by default. This diverges from the behavior of the Windows build script. My reason for diverging here was because, without having users specify an install directory, I felt that Linux devs, in general, would associate `make install` with moving the files to common install locations (e.g., moving `$PWD/build/eoserv` to `/usr/bin/eoserv`). 

This change in the CMake file highlights where this dichotomy can be a problem:
- A Windows user will use the default command (build & install) and the `local_conf` directory will be present in the install directory.
- A Linux user will use the default command (build) and the `local_conf` directory will not be present. The Linux user will then need to copy the `local_conf`.

I'm fine with reverting back the Linux behavior to build and install by default. I have the change made locally for this branch, so if that's what we want to do then I'll push the change.

However, before we make that decision can we quickly discuss the deployment workflow? For the instance you are currently serving, did you just deploy it from the `install` directory?

Reply:
Hmmm that's a good point. I think it is smart to decouple deployment from the CMake concept of installing. My current Ubuntu deployment process is:

1. git pull
2. ./build-linux.sh -i
3. restart service

The reason I made this change was because I had all my production files in install (where it was executing from) and did a clean build without thinking (due to other build errors), so I had to recreate the configs. Now I've got my local configs in `config_local` and wanted those to automatically get copied to the install directory. For anyone else building, I think this makes sense.

As for doing build/install automatically, I think this should be done the same on both platforms. IMO the goal of a build is to generate some artifact that is ready to use, which is what the `install` target does right now. If there's a better way to do this in CMake then I'm open to that.

Reply:
The only way we could make the build step completely useful is by copying all the files there first (i.e., replacing `install` calls with `add_custom_command` using cmake copy). However, that would be tedious and push the dichotomy of the workflows further away.

Therefore, I'll commit and push my local change that unifies the workflows so that the Linux build script both builds and installs by default. This means your Ubuntu workflow will become:

```bash
git pull
./build-linux.sh
restart service
```

Reply:
@ethanmoffat I'll leave this conversation for you to resolve to make sure you see my last comment.